frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -843,6 +843,29 @@ function ReaderHighlight:onHoldRelease()
     return true
 end
 
+function ReaderHighlight:onToggleHighlightAction()
+    local actions = {
+        highlight = "translate",
+        translate = "wikipedia",
+        wikipedia = nil
+    }
+    local current_action = G_reader_settings:readSetting("default_highlight_action")
+    if not current_action then
+        G_reader_settings:saveSetting("default_highlight_action", "highlight")
+        UIManager:show(InfoMessage:new{
+            text = _("Default highlight action changed to highlight"),
+            timeout = 1,
+        })
+    else
+        local next_action = actions[current_action]
+        G_reader_settings:saveSetting("default_highlight_action", next_action)
+        UIManager:show(InfoMessage:new{
+            text = _("Default highlight action changed to " .. (next_action or "default")),
+            timeout = 1,
```

Comment:
And you might want to see if you'd not prefer `Notification` instead of `InfoMessage`, which makes thing less busy with rereshes. It's possible that this text is too long for a Notification, that may depends on screen dpi (haven't checked if long text is wrapped).

Reply:
Notifications seem to work nicely and don't depend on DPI, at least on emulator, I'll test it on my kindle later and possibly amend this commit. For now everything seems to work without any problems.